app/js/Banner.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+let paper = require('js/paper-core.min'),
+    paperScope = require('js/PaperScopeManager');
+
+class Banner {
+
+    /**
+     * @param iconBase of the icon used to cut the shadow
+     */
+
+    constructor(iconPath, iconBase) {
+        this.iconBase = iconBase;
+        this.iconPath = iconPath;
+        this.bannerBeta = false;
+    }
+
+    setBannerBeta(boolean){
+      this.bannerBeta = boolean;
+    }
+
+    applyBanner(){
```

Comment:
`Icon` and `IconShadow` have apply methods which  intersect the icon / shadow with base below. Banner does not need this, right? So let's move the construction of the banner to the constructor and remove this method.